From: Jakub Adam <jakub.adam@ktknet.cz>
Date: Thu, 5 Jan 2012 00:57:49 +0100
Subject: probe-lib-location

Install gammaray_probe.so and libgammaray_widget_export_actions.so
to /usr/lib/gammaray as it is application private library.
Fixes lintian warnings/errors:

package-name-doesnt-match-sonames
shlib-without-versioned-soname
postinst-must-call-ldconfig
---
 CMakeLists.txt                       |    5 +++--
 config-gammaray.h.cmake              |    2 +-
 injector/styleinjector.cpp           |    2 +-
 tools/widgetinspector/CMakeLists.txt |    2 +-
 4 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 06bdc1c..bc69cd0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,6 +87,7 @@ include_directories(
 set(LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)")
 set(LIB_INSTALL_DIR "lib${LIB_SUFFIX}")
 set(PLUGIN_INSTALL_DIR "${LIB_INSTALL_DIR}/qt4/plugins")
+set(MODULE_INSTALL_DIR "${LIB_INSTALL_DIR}/gammaray")
 
 set(
   INSTALL_TARGETS_DEFAULT_ARGS
@@ -314,7 +315,7 @@ endif()
 
 qt4_add_resources(gammaray_srcs resources/gammaray.qrc)
 add_library(gammaray_probe
-  SHARED ${gammaray_srcs}
+  MODULE ${gammaray_srcs}
 )
 if(BUILD_TIMER_PLUGIN AND NOT WIN32)
   target_link_libraries(gammaray_probe rt)
@@ -328,7 +329,7 @@ if(NOT WIN32)
 endif()
 set_target_properties(gammaray_probe PROPERTIES PREFIX "")
 
-install(TARGETS gammaray_probe ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS gammaray_probe DESTINATION ${MODULE_INSTALL_DIR})
 
 if(UNIX AND NOT APPLE)
   set(XDG_APPS_INSTALL_DIR share/applications)
diff --git a/config-gammaray.h.cmake b/config-gammaray.h.cmake
index e1a24d2..c6c8b89 100644
--- a/config-gammaray.h.cmake
+++ b/config-gammaray.h.cmake
@@ -1,4 +1,4 @@
-#define GAMMARAY_LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}"
+#define GAMMARAY_LIB_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/${LIB_INSTALL_DIR}/gammaray"
 #define GAMMARAY_PLUGIN_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/${PLUGIN_INSTALL_DIR}"
 // for finding the probe during automatic tests
 #define GAMMARAY_BUILD_DIR "${CMAKE_BINARY_DIR}"
diff --git a/injector/styleinjector.cpp b/injector/styleinjector.cpp
index 3d23a12..6b02dc1 100644
--- a/injector/styleinjector.cpp
+++ b/injector/styleinjector.cpp
@@ -94,7 +94,7 @@ bool StyleInjector::selfTest()
   // TODO: be a bit more clever in finding the plugin location (also when actually using it above)
 #ifndef Q_OS_WIN
   const QString stylePath =
-    QLatin1String(GAMMARAY_LIB_INSTALL_DIR "/qt4/plugins/styles/gammaray_injector_style.so");
+    QLatin1String(GAMMARAY_PLUGIN_INSTALL_DIR "/styles/gammaray_injector_style.so");
 #else
   const QString stylePath =
     QLatin1String(GAMMARAY_LIB_INSTALL_DIR "/qt4/plugins/styles/gammaray_injector_style.dll");
diff --git a/tools/widgetinspector/CMakeLists.txt b/tools/widgetinspector/CMakeLists.txt
index eefa624..e851f8d 100644
--- a/tools/widgetinspector/CMakeLists.txt
+++ b/tools/widgetinspector/CMakeLists.txt
@@ -10,4 +10,4 @@ target_link_libraries(gammaray_widget_export_actions
   ${QT_QTDESIGNER_LIBRARIES}
 )
 
-install(TARGETS gammaray_widget_export_actions ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS gammaray_widget_export_actions DESTINATION ${MODULE_INSTALL_DIR})
-- 
